dhcp-relay/dhcp_kern_xdp.c

```diff
@@ -285,23 +378,20 @@ int xdp_dhcp_relay(struct xdp_md *ctx) {
 
 	/* Read device name from device map */
 	key = 0;
-	dev = bpf_map_lookup_elem(&device_name, &key);
-	if (dev == NULL)
+	dev_config = bpf_map_lookup_elem(&device_name, &key);
+	if (dev_config == NULL)
 		goto out;
 
-	/* Copy headers of packet to buf */
-	//if (xdp_load_bytes(ctx, 0, buf, static_offset))
-	//	goto out;
+	memcpy(dev.name, dev_config, RAI_OPTION_LEN);
+	dev.len = str_len(&dev);
```

Comment:
This is just looping through the string to find the length of it, only to use that length as the end condition of a loop later. This is totally superfluous, just check for the null terminator directly in the loop that uses length and get rid of this call (and lose the `len` member of struct dev_name entirely).